fastserde/avro-fastserde-tests-common/src/test/java/com/linkedin/avro/fastserde/logical/types/InMemoryEncoder.java

```diff
@@ -13,6 +13,17 @@
 import com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper;
 import com.linkedin.avroutil1.compatibility.SchemaNormalization;
 
+/**
+ * Helper class which wraps {@link BinaryEncoder} and its corresponding {@code OutputStream}
+ * ({@link ByteArrayOutputStream}) so that we can simplify some repeatable code just to:
+ * <pre>
+ * InMemoryEncoder encoder = new InMemoryEncoder(data.getSchema());
+ * fastSerializer.serialize(data, encoder);
+ * return encoder.toByteArray();
+ * </pre>
+ * Additionally, it prepends 10-bytes header to the output byte-array so that it can be easily compared with
+ * built-in serialization method: {@code someSpecificRecord.toByteBuffer().toByteArray()}.
```

Comment:
So... is the point of the header that two records of different schemas that happen to encode to the same bytes will not be mistakenly treated as equal?